shelly/src/main/java/au/com/jtribe/shelly/Map.java

```diff
@@ -0,0 +1,160 @@
+package au.com.jtribe.shelly;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.support.annotation.CheckResult;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import static au.com.jtribe.shelly.Preconditions.checkNotNull;
+
+/**
+ * Represents a Map which can be set to a specific address or coordinate (lat, long)
+ */
+public final class Map {
+
+  //Can use either lat/long combination or address
+  private Double latitude;
+  private Double longitude;
+  private String address;
+
+  private Integer zoom;
+  //Would be good to implement this at some point
+  //private String label;
+
+  Map() {
+  }
+
+  /**
+   * Adds a latitude to the Map
+   * Expects a value between 180 and -180.
+   *
+   * @param latitude The latitude coordinate for the Map
+   * @return Object this method was called on for method chaining.
+   */
+  @NonNull
+  @CheckResult
+  public Map latitude(@NonNull Double latitude) {
```

Comment:
situations like here I think using the boxed value is fine because we _need_ the user to specify both the latitude and longitude.